### PR TITLE
CT Alignment Geometry

### PR DESCRIPTION
**Pull request by SergejMuhic**
Modified the RuleIDs and added Identifier and Type restrictions.

### CODE REVIEW

**Review by AlexBrad1eyCT — CHANGES_REQUESTED** [review]
issues with the constraints and referencing them

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Base for a standard two (Alignment Geometry Gradient) and three (Alignment Geometry Gradient) layout alignment geometric representation.
```
Comment (pjanck) [inline IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md:1]: It states `Alignment Geometry Gradient` twice - intentional?

**Review by pjanck — CHANGES_REQUESTED** [review]
Additional question to Alex's suggestions and questions.

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md (hunk 1 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md:1]: I think it should be:
```suggestion
Base for a standard two (Alignment Geometry Gradient) and three (Alignment Geometry Cant) layout alignment geometric representation.
```

maybe @SergejMuhic

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml [hunk 2]
```diff
@@ -18,12 +18,31 @@
...
+														<DocModelRuleConstraint>
+															<Expression xsi:type="DocOpStatement" Operation="compareequal">
+																<Reference xsi:type="DocOpReference">
+																	<EntityRule xsi:type="DocModelRuleEntity" xsi:nil="true" href="IfcLabel_1FlfrpXn6XWDZbdkjPKg" />
+																</Reference>
+																<Value xsi:type="DocOpLiteral" Operation="loadstring" Literal="Curve3D" />
+															</Expression>
+														</DocModelRuleConstraint>
+													</Rules>
+												</DocModelRuleEntity>
```
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml:40]: Yes, Forgot to re-serialize this after the applied IfcDoc fix. This just got screwed over with the previous serialization.

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md (hunk 1 above)
Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md:1]: I will look into this.

Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Documentation.md]: Addressed.

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml:40]: Serialization is still of. Have to fix that first.

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Cant/DocTemplateDefinition.xml [hunk 3]
```diff
@@ -18,12 +18,12 @@
 										</DocModelRuleAttribute>
 										<DocModelRuleAttribute CardinalityMin="1" CardinalityMax="1" Name="RepresentationIdentifier" Identification="Identifier">
 											<Rules>
-												<DocModelRuleEntity Name="IfcLabel" />
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_1FlfrpXn6XWDZbdkjPKg" />
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Cant/DocTemplateDefinition.xml:21]: Correct

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml [hunk 4]
```diff
@@ -18,12 +18,12 @@
 										</DocModelRuleAttribute>
 										<DocModelRuleAttribute CardinalityMin="1" CardinalityMax="1" Name="RepresentationIdentifier" Identification="Identifier">
 											<Rules>
-												<DocModelRuleEntity Name="IfcLabel" />
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_1FlfrpXn6XWDZbdkjPKg" />
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:21]: Correct

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml [hunk 5]
```diff
@@ -18,12 +18,12 @@
...
 											</Rules>
 										</DocModelRuleAttribute>
 										<DocModelRuleAttribute Name="RepresentationType" Identification="Type">
 											<Rules>
-												<DocModelRuleEntity Name="IfcLabel" />
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_1FlfrpXn6XWDZbdkjPKg" />
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:26]:
```suggestion
												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_34VbbuccrDUuUE77e2rJVV" />
```

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Cant/DocTemplateDefinition.xml [hunk 6]
```diff
@@ -18,12 +18,12 @@
...
 											</Rules>
 										</DocModelRuleAttribute>
 										<DocModelRuleAttribute Name="RepresentationType" Identification="Type">
 											<Rules>
-												<DocModelRuleEntity Name="IfcLabel" />
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_1FlfrpXn6XWDZbdkjPKg" />
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Cant/DocTemplateDefinition.xml:26]:
```suggestion
												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_34VbbuccrDUuUE77e2rJVV" />
```

**Review by AlexBrad1eyCT — APPROVED** [review]
Serialization corrected just a hang over from the previous bugs with wrong href
Just need the corrections suggested then LFTM

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml]: Fixed via comment suggestion

**Review by AlexBrad1eyCT — APPROVED** [review]
This seems all find to me know! Lets get it merged!

**Comment by AlexBrad1eyCT** [comment]
@pjanck @jmirtsch @czapplitec @apinzenoehler If you would like to review please do asap, preferably by end of day wednesday, so we can get it merged.

**Pull request merged**